src/app/issues-viewer/issues-viewer.component.ts

```diff
@@ -106,53 +99,9 @@ export class IssuesViewerComponent implements OnInit, AfterViewInit, OnDestroy {
       this.availableGroupsSubscription.unsubscribe();
     }
 
-    // Fetch assignees
-    this.groups = [];
-    this.hiddenGroups = [];
-
-    this.availableGroupsSubscription = this.groupingContextService.getGroups().subscribe((x) => (this.groups = x));
-  }
-
-  /**
-   * Checks if our current repository available on view service is indeed a valid repository
-   */
-  private checkIfValidRepository() {
-    const currentRepo = this.viewService.currentRepo;
-
-    if (Repo.isInvalidRepoName(currentRepo)) {
-      return of(false);
-    }
-
-    return (
-      (this.githubService.isOrganisationPresent(currentRepo.owner) || this.githubService.isUsernamePresent(currentRepo.owner)) &&
-      this.githubService.isRepositoryPresent(currentRepo.owner, currentRepo.name)
-    );
-  }
-
-  /**
-   * Update the list of hidden group based on the new info.
-   * @param issueLength The number of issues under this group.
-   * @param target The group.
-   */
-  updateHiddenGroups(issueLength: number, target: Group) {
-    // If the group is in the hidden list, add it if it has no issues.
-    // Also add it if it is unchecked in the filter.
-    if (issueLength === 0 && this.groupingContextService.isInHiddenList(target)) {
-      this.addToHiddenGroups(target);
-    } else {
-      this.removeFromHiddenGroups(target);
-    }
-  }
-
-  private addToHiddenGroups(target: Group) {
-    const isGroupPresent = this.hiddenGroups.some((group) => group.equals(target));
-
-    if (!isGroupPresent) {
-      this.hiddenGroups.push(target);
-    }
-  }
+    this.groupService.groups = [];
+    this.groupService.hiddenGroups = [];
 
-  private removeFromHiddenGroups(target: Group) {
-    this.hiddenGroups = this.hiddenGroups.filter((group) => !group.equals(target));
+    this.availableGroupsSubscription = this.groupingContextService.getGroups().subscribe((x) => (this.groupService.groups = x));
```

Comment:
After moving the dependency of  GroupingContextService to GroupService, this line can also be encapsulated with line 102-103 as one method.